ReactSkia/sdk/ThreadSafeCache.h

```diff
@@ -0,0 +1,58 @@
+/*
+* Copyright (C) 1994-2022 OpenTV, Inc. and Nagravision S.A.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+#include <mutex>
+#include <better/map.h>
+#include <better/optional.h>
+#include "ReactSkia/utils/RnsLog.h"
+
+using namespace std;
+namespace rns {
+namespace sdk {
+using namespace facebook;
+// KeyT any type and ValueT any type
+// RSkThreadSafeCache<<key>,<value>>
+//RSkThreadSafeCache<string,shared_ptr<CurlResponse>>
+template <typename KeyT, typename ValueT>
+class RSkThreadSafeCache {
+ private:
+  static  int overallCurrentSize_;
+  static  int overallMaxSize_;
+  std::mutex cacheLock_;
+  int currentSize_;
+  int maxSize_;
+  KeyT key_;
+  better::map <KeyT,ValueT> cacheMap_;
+ public:
+  bool findCache(const char* url){
```

Comment:
What is purpose of this this function ?  This cannot be non-generic right ? May be you meant to use KeyT for searching ? 
Either use name isAvailableinCache(keyT) or We can re-purpose getCacheData() for checking also